packages/apputils/src/sessioncontext.tsx

```diff
@@ -345,6 +348,12 @@ export namespace ISessionContext {
      */
     settingRegistry?: ISettingRegistry | null;
   }
+  /**
+   * On before restarting the kernel options
+   */
+  export interface IRestartOptions {
+    onBeforeRestart: () => Promise<void>;
```

Comment:
Ideally the methods in the interface would be documented too.

Reply:
@krassowski Thanks for the comment. I have added a description, hoping it is ok.